### DISCUSSION TITLE
Repository not onboarded when added to Github App

### DISCUSSION THREAD

**Discussion by theurichde** [renovatebot/renovate#23502] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Hi there!

When we add repositories to the GitHub App, some repositories don't get onboarded to Renovate. They are visible in the Mend Renovate Dashboard, but they don't get the initial onboarding job triggered.

We add the repositories via [GitHub API](https://docs.github.com/en/rest/apps/installations?apiVersion=2022-11-28#add-a-repository-to-an-app-installation) with a grace period of 2 seconds between each repository. (If you wonder why: we enable teams to add their team slug in the automation, and we fetch all their (sometimes > 50) repos)

Is there anything up with the GitHub App? WebHook Rate Limiting or some bad processing on your side? Can we do something on our side?

Here are some example repositories which are installed, but have no initial onboarding job run yet.

* https://developer.mend.io/github/idealo/ibt-category-repository
* https://developer.mend.io/github/idealo/ibt-content-repository
* https://developer.mend.io/github/idealo/ibt-datalake-processing
* https://developer.mend.io/github/idealo/ibt-product-repository
* https://developer.mend.io/github/idealo/ibt-report-service
* https://developer.mend.io/github/idealo/ibt-shop-repository

Thanks!

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by theurichde** [comment]
This problem still persists 😢

**Reply by theurichde** [reply]
@rarkins @PhilipAbed The problem is still there and occurs for new repositories from time to time.

**Reply by PhilipAbed** [reply]
Hi @theurichde i believe it might be a configuration in your org/repo settings.
are you sure renovate is enabled in settings?
are you sure you set your org to have onboarding prs?


my second question is, did you add multiple repositories 1 by 1 ? instead of marking them all and adding them?

**Reply by theurichde** [reply]
Hey there 👋🏻 

1.a) We have already onboarded 690 repositories, almost all are working fine.
1.b) this includes onboarding PRs

2.) We add multiple repositories one by one programmatically with a grace period of 2 seconds between each addition.

**Reply by rarkins** [reply]
We trigger an org sync when we see a new repo so I'm not sure how it would behave if they're added so closely together

**Reply by theurichde** [reply]
Okay, got that. 

I try increasing the grace period between additions. Any recommendations? I think a human can add and save one single repository in about 5 seconds at a time 🤔 

In the meantime, there were some new repositories added. Shouldn't that trigger the sync and enable / trigger Renovate for those repositories, too?

**Reply by PhilipAbed** [reply]
sync will sync the new repositories, not the jobs.
once adding a repository fails to add a new job, it will be ignored.

**Workaround**: create a new job using the "Run Renovate" button manually for each failed repo


Regarding the Grace period, if it was me with . i would rather wait as much as possible, 
can you send me the most recent repository to investigate?

**Reply by theurichde** [reply]
> once adding a repository fails to add a new job, it will be ignored.
Ah, okay, good to know!

The workaround works fine 🌻 

@PhilipAbed I _believe_ this is one of the most recent ones. → https://developer.mend.io/github/idealo/ibt-product-repository
I didn't stumble upon new ones yet (e.g. being added 24 hours ago)

(off-topic sidenote: You folks are doing a great job, and we really appreciate and enjoy Renovate!)
